### ISSUE TITLE
Client: URL encode / escaping is wrong

### ISSUE THREAD

**Issue by sobotkami** [ktorio/ktor#1945]
**Ktor Version and Engine Used (client or server and name)**
Version: 1.3.2-1.4-M1-2
ktor-client-core
ktor-client-json-native
ktor-client-serialization-native
ktor-client-curl

**Describe the bug**
The URL is wrong escaped.

**To Reproduce**
```kotlin
class FahrplanApi(private val serverUrl: String = "http://api.deutschebahn.com/freeplan/v1") {
    private val client = HttpClient {
        install(JsonFeature)
        // workaround for version 1.4-M1 
        // https://github.com/ktorio/ktor/issues/1820#issuecomment-633519483
        HttpResponseValidator {
            validateResponse { response ->
                when (response.status.value) {
                    in 300..399 -> throw RedirectResponseException(response)
                    in 400..499 -> throw ClientRequestException(response)
                    in 500..599 -> throw ServerResponseException(response)
                }
            }
        }

        expectSuccess = false
    }

    suspend fun departureBoardDateTime(id: Long, date: String) =
        client.get<Departure>("$serverUrl/departureBoard/$id?date=$date")
}

fun main(): Unit = runBlocking {
    val api = FahrplanApi()
    api.departureBoardDateTime(8000096, "2020-06-14T20:21:22")
}
```

after run:
```
Uncaught Kotlin exception: io.ktor.client.features.ClientRequestException: Client request(https://api.deutschebahn.com/freeplan/v1/departureBoard/8000096?date=2020-06-14T20%253A21%253A22) invalid: 400 Bad Request
```

**Expected behavior**
```
https://api.deutschebahn.com/freeplan/v1/departureBoard/8000096?date=2020-06-14T20%3A21%3A22
```
`%253A` --> `%3A` (25 is actually `%`)

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-341">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Closed. Fixed in `1.4.1`. Added tests
